frontend/src/lib/fabricCanvasUtils.ts

```diff
@@ -0,0 +1,364 @@
+    const ctx = canvas.getContext('2d');
+    ctx?.drawImage(img, 0, 0, width, height);
+    const resizedImage = canvas.toDataURL();
+    callback(resizedImage);
+  };
+};
+
+export const handleZoom = (opt: fabric.IEvent<WheelEvent>): void => {
+  const evt = opt.e;
+  const target = opt.target as unknown as fabric.Canvas | undefined;
+  if (target && target instanceof fabric.Canvas) {
+    const delta = evt.deltaY;
+    const pointer = target.getPointer(evt);
+    const zoom = target.getZoom();
+    const newZoom = zoom * (1 - delta / 200);
+    target.zoomToPoint(new fabric.Point(pointer.x, pointer.y), newZoom);
+    evt.preventDefault();
+    evt.stopPropagation();
+  }
+};
+
+export const saveImagesToLocalFile = (canvas: fabric.Canvas | null) => {
+  if (canvas) {
+    const images = canvas.getObjects('image').map((img) => {
+      const image = img as fabric.Image;
+      return {
+        type: 'image',
+        left: image.left,
+        top: image.top,
+        width: image.width,
+        height: image.height,
+        scaleX: image.scaleX,
+        scaleY: image.scaleY,
+        angle: image.angle,
+        src: image.getSrc(),
+      };
+    });
+
+    const jsonContent = JSON.stringify(images, null, 2);
+    const blob = new Blob([jsonContent], { type: 'application/json' });
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = 'canvas_data.json';
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
+    URL.revokeObjectURL(url);
+  }
+};
+
+export const loadImagesFromJSON = (canvas: fabric.Canvas | null, json: string) => {
+  if (canvas) {
+    const images: CanvasImage[] = JSON.parse(json);
+
+    images.forEach((imgData) => {
+      fabric.Image.fromURL(imgData.src, (img) => {
+        img.set({
+          left: imgData.left,
+          top: imgData.top,
+          scaleX: imgData.scaleX,
+          scaleY: imgData.scaleY,
+          angle: imgData.angle,
+        });
+        canvas.add(img);
+      });
+    });
+
+    canvas.renderAll();
+  }
+};
+
+
+export const getSelectedObjectStyles = (canvas: fabric.Canvas | null): object | null => {
+  if (canvas) {
+    const activeObject = canvas.getActiveObject();
+    if (activeObject) {
+      return activeObject.toObject([
+        'fill', 'stroke', 'strokeWidth', 'fontSize', 'fontFamily', 'fontWeight', 'fontStyle', 'textAlign', 'angle', 'left', 'top', 
+      ]);
+    }
+  }
+  return null;
+};
+
+
+export const setSelectedObjectStyles = (canvas: fabric.Canvas | null, styles: object): void => {
+  if (canvas) {
+    const activeObjects = canvas.getActiveObjects();
+    activeObjects.forEach((obj) => {
+      obj.set(styles);
+    });
+    canvas.requestRenderAll();
+  }
+};
+
+export const updateDimensions = (obj: any) => {
+  if (!obj) {
+    return;
+  }
+  
+  if (obj.type === 'rect' || obj.type === 'circle' || obj.type === 'line' || obj.type === 'path') {
```

Comment:
Maybe we should separate circles with radiuses from other shapes (without radiuses)?

Reply:
Actually that line was creating a bug, rect, circle, line behave in different way than line and image - line and image cannot be scaled in this way.

Reply:
You know, never mind. My proposition of change is not really needed here